distributed/deploy/spec.py

```diff
@@ -271,7 +271,7 @@ async def _start(self):
         self._lock = asyncio.Lock()
         self.status = Status.starting
 
-        if self.scheduler_spec is None:
```

Comment:
keep this, please. This smells like dead code or something we _should_ be testing